src/features/categories/category-selection/CategorySelect.tsx

```diff
@@ -31,52 +33,62 @@ const CategorySelect = ({
     setIsOpen((prev) => !prev);
   };
 
-  const handleCategorySelection = (categoryId: IndexableType, operation: 'add' | 'remove') => {
-    switch (operation) {
-      case 'add':
-        setSelectedCategories([...categoryIds, categoryId]);
-        break;
-      case 'remove':
-        setSelectedCategories(categoryIds.filter((id) => id !== categoryId));
-        break;
-    }
-  };
+const handleCategorySelection = (categoryId: IndexableType, operation: 'add' | 'remove') => {
+  if (readOnly && operation === 'remove') return;
```

Comment:
This `if` can be removed since you only assign the callback in case it's not `readOnly` mode.